### PR TITLE
ENH : CSP and Xdawn share code

### PR DESCRIPTION
**Pull request by alexandrebarachant**
This is a PR to factorize shared code between Xdawn and CSP. (cf #2296 )
The shared method is _regularized_covariance.

### CODE REVIEW

File: mne/decoding/csp.py [hunk 1]
```diff
@@ -1,17 +1,93 @@
...
+        else:
+            raise ValueError("regularization parameter should be "
+                             "of type str or int (got %s)." % type(reg))
+
+        # compute regularized covariance using sklearn
+        cov = skl_cov.fit(data.T).covariance_
+
+    return cov
+
+
```
Comment (Eric89GXL) [inline mne/decoding/csp.py]: My feeling is that this should actually live in `mne/cov.py`, @agramfort do you agree?

Comment (agramfort) [inline mne/decoding/csp.py]: Indeed sorry for bad suggestion

**Comment by alexandrebarachant** [comment]
I moved the code in cov.py.
I also took care of a few pep8 warning (my flake8 enhanced editor was complaining about missing period.)

Do you want a test with that or you thing the CSP + Xdawn coverage is enough

**Comment by agramfort** [comment]
Thanks!

**Pull request merged**